controller/grpc/controller.proto

```diff
@@ -0,0 +1,275 @@
+syntax = 'proto3';
+package controller;
+option go_package = 'controllergrpc';
+
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/any.proto";
+
+service Controller {
+  rpc ListApps (ListAppsRequest) returns (ListAppsResponse) {}
+  rpc GetApp (GetAppRequest) returns (App) {
+    option (google.api.http).get = "/v2/{name=apps/-}";
+  };
+  rpc StreamAppLog (StreamAppLogRequest) returns (stream LogChunk) {
+    option (google.api.http).get = "/v2/{name=apps/-}/log}";
+  };
+  rpc CreateRelease (CreateReleaseRequest) returns (Release) {
+    option (google.api.http).post = "/v2/{parent=apps/-}/releases}";
+  };
+  rpc CreateDeployment (CreateDeploymentRequest) returns (Deployment) {
+    option (google.api.http).post = "/v2/{parent=apps/-}/deployments";
+  };
+  rpc StreamEvents (StreamEventsRequest) returns (stream Event) {
+    option (google.api.http).get = "/v2/events";
+  };
+  // TODO(jvatic): rpc ListAppRoutes
+  // TODO(jvatic): rpc CreateAppRoute
+  // TODO(jvatic): rpc UpdateAppRoute
+  // TODO(jvatic): rpc DeleteAppRoute
+}
+
+message ListAppsRequest {
+  int32 page_size = 1;
+  string page_token = 2;
+}
+
+message ListAppsResponse {
+  repeated App apps = 1;
+  string next_page_token = 2;
+}
+
+message GetAppRequest {
+  // name = "apps/APP_ID"
+  string name = 1;
+}
+
+message StreamAppLogRequest {
+  // name = "apps/APP_ID"
+  string name = 1;
+  LogAggregatorLogOpts opts = 2;
+}
+
+message CreateReleaseRequest {
+  // parent = "apps/APP_ID"
+  string parent = 1;
+  Release release = 2;
+  string request_id = 3;
+}
+
+message CreateDeploymentRequest {
+  // parent = "apps/APP_ID"
+  string parent = 1;
+  // release = Release.name
+  string release = 2;
+  string request_id = 3;
+}
+
+message StreamEventsRequest {
+  // parent = "apps/APP_ID"
+  string parent = 1;
+  repeated string object_types = 2; // []ct.EventType
+  string object_id = 3;
+  bool post = 4;
+  int32 count = 5;
+}
+
+message App {
+  // name = "apps/APP_ID"
+  string name = 1;
+  string display_name = 2;
+  map<string, string> labels = 3;
+  google.protobuf.Timestamp create_time = 4;
+  google.protobuf.Timestamp update_time = 5;
+  int32 deploy_timeout = 6;
+  string strategy = 7;
+  // release = Release.name
+  string release = 8;
+}
+
+message Release {
+  // name = "apps/APP_ID/releases/RELEASE_ID"
+  string name = 1;
+  repeated string artifacts = 3;
+  map<string, string> env = 4;
+  map<string, string> labels = 5;
+  map<string, ProcessType> processes = 6;
+  google.protobuf.Timestamp create_time = 7;
+}
+
+message Deployment {
+  // name = "apps/APP_ID/deployments/DEPLOYMENT_ID"
+  string name = 1;
+  // old_release = Release.name
+  string old_release = 3;
+  // new_release = Release.name
+  string new_release = 4;
+  string strategy = 5;
+  string status = 6;
```

Comment:
I think this should be an enum?

Reply:
It's not one in the controller, but looks like it should be. I'll create one for it here and make a note to refactor it into one in the controller.